### DISCUSSION TITLE
How to change the trigger time through csv

### DISCUSSION THREAD

**Discussion by karthikkumarstp1** [Azure-Player/azure.datafactory.tools#193] [Q&A] [answered]
Hello,

I have a following trigger in my ADF

`{
    "name": "TempoTrigger",
    "properties": {
        "annotations": [],
        "runtimeState": "Stopped",
        "pipelines": [
            {
                "pipelineReference": {
                    "referenceName": "Tempo",
                    "type": "PipelineReference"
                }
            }
        ],
        "type": "ScheduleTrigger",
        "typeProperties": {
            "recurrence": {
                "frequency": "Day",
                "interval": 1,
                "startTime": "2021-05-05T05:42:00",
                "endTime": "2022-05-12T10:29:46",
                "timeZone": "India Standard Time",
                "schedule": {
                    "minutes": [
                        0
                    ],
                    "hours": [
                        7
                    ]
                }
            }
        }
    }
}`

I need to change the hours property from "7" to "8" My config.csv is like this 

`trigger,TempoTrigger,typeProperties.recurrence.schedule.hours,'[8]'`
also tried this
`trigger,TempoTrigger,typeProperties.recurrence.schedule.hours,'8'`
`trigger,TempoTrigger,typeProperties.recurrence.schedule.hours,"8"`
`trigger,TempoTrigger,typeProperties.recurrence.schedule.hours,8`

But it is not updating the triggers. Any better way to write this in config.csv.

Thanks in advance

**Comment by NowinskiK** [comment, marked as answer]
Hi @karthikkumarstp1, thanks for raising the question. A very good example, which is not covered by the documentation.
Finally, I had a chance to check that out for you and you can do this in two ways:

**Option 1**
Replace the first (index=0) item in the array ( [ ] represent an array )
`trigger,TempoTrigger,typeProperties.recurrence.schedule.hours[0],8`
This is a bit risky and I do NOT recommend that approach.

**Option 2 (recommended)**
Replace the whole `schedule` element.
`trigger,TempoTrigger,typeProperties.recurrence.schedule,{"minutes": [0],"hours": [7]}`

**Reply by karthikkumarstp1** [reply]
Hello Kamil, Thanks for the response. I tried option 2 and I am getting this error.

_ADFT0025: Config file, row 1 has too many columns._

**Reply by NowinskiK** [reply]
My bad, sorry. 
It's CSV (Comma Separated Values), and because of the value:
1) contains a comma - the whole value must be encapsulated with `"` (double quote)
2) contains double quotes - all double quotes must be enquoted (`"` -> `""`)

```
type,name,path,value
trigger,TempoTrigger,typeProperties.recurrence.schedule,"{""minutes"": [0],""hours"": [7]}"
```

This should work now, HTH and makes sense.

**Comment by JADawson** [comment]
Hi Kamil - I'm having a similar problem trying to set the "Hour" on a timer trigger.

I have tried the following, based on the recommendations above

`trigger,trDataTransferOldDBtoNewDB,typeProperties.recurrence.schedule.hours[0],21`

and

`trigger,trDataTransferOldDBtoNewDB,typeProperties.recurrence.schedule, "{""minutes"": [0],""hours"": [21]}"`

Both of these are giving me an error "**Unable to deserialize the response**." when the release pipeline executes...

**Reply by NowinskiK** [reply]
James, now you have the second, different problem: "Unable to deserialize the response".
This happens always when #adftools tries to read an object from ADF (target) instance and object is invalid. 
In your case, the trigger became invalid during one of the previous deployments.
To fix the issue, drop the trigger manually from UI and rerun the release pipeline.

**Comment by kishorekuma** [comment]
Hi Nowinski,

Below is json file format  for my scheduled trigger.

{
    "name": "trigger_1",
    "properties": {
        "annotations": [],
        "runtimeState": "Stopped",
        "pipelines": [],
        "type": "ScheduleTrigger",
        "typeProperties": {
            "recurrence": {
                "frequency": "Week",
                "interval": 1,
                "startTime": "2022-07-13T12:59:00Z",
                "timeZone": "UTC",
                "schedule": {
                    "weekDays": [
                        "Monday",
                        "Tuesday",
                        "Wednesday",
                        "Thursday",
                        "Friday"
                    ]
                }
            }
        }
    }
}
I have tried following your recommendations in creating csv file 

type,name,path,value
trigger,trigger_1,typeProperties.recurrence.frequency.interval.startTime.timeZone.schedule.weekDays,"{""week"",""1"",""2022-07-13T12:59:00Z"",""UTC"",""Monday,Tuesday,Wednesday,Thursday,Friday""}"
Still i am getting an error saying wrong path defined.
Can you please help me configure the scheduled trigger with frequencies of days , weeks and months.

**Reply by NowinskiK** [reply]
Your path is wrong. `Schedule` is child for `recurrence`, NOT for `timeZone` (which is only another child of `recurrence`)
![image](https://user-images.githubusercontent.com/25029325/192271143-0fb1db13-d6dc-4519-97e7-ac148237f833.png)
Your path should be: **typeProperties.recurrence.schedule** and then define everything what should be there.
